### PR TITLE
feat: camp overview information

### PR DESCRIPTION
**Pull request by amanda-guo**
## Notion ticket link

[Camp Overview: Camp level information view only](https://www.notion.so/uwblueprintexecs/Camp-Overview-Camp-level-information-view-only-343f864f624945edacc3d6fe599c0ebf)

## Implementation description
* Created a camp overview page that just includes the top-level camp information according to the Figma designs
* Added an APIClient to call the BE APIs to get the camp information
* There are some aspects of this overview that are out of scope of this ticket and some things that are blocked too (check Additional Notes & Clarifications in the Notion ticket to see what's included and not included)
* Currently a camp is just hard coded for now
* Some "TODOs" have been called out as comments in the code (left those there as placeholders for future updates that were deemed out-of-scope for this ticket)

## Steps to test
1. Build containers and open the page at http://localhost:3000/admin/camp/62c098e7b4a7a433a7622ff4 on a browser
2. Check that this matches the Figma designs and that the right information is pulled from the DB

## What should reviewers focus on?
* Code quality and placement

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/App.tsx
Comment (dhmi19) [inline frontend/src/App.tsx:109]: Why do we need to replace the ID parameter here? I tried removing the .replace() function and then reloading the page with the given url and it still worked. Removing this here would save us the refactor in the future maybe?

code:
`<PrivateRoute
                  exact
                  path={Routes.CAMP_OVERVIEW_PAGE}
                  component={CampOverview}
                />`

URL tested: http://localhost:3000/admin/camp/62c098e7b4a7a433a7622ff4

File: frontend/src/constants/AuthConstants.ts
Comment (dhmi19) [inline frontend/src/constants/AuthConstants.ts:5]: Was this part of another ticket? If so, just wanted to make sure this is not gonna create any conflicts on merge?

File: frontend/src/types/CampsTypes.ts
Comment (dhmi19) [inline frontend/src/types/CampsTypes.ts:10]: Is it ok for these fields to be strings? In the future, we'll need to create 30 minute timeslots from Early Dropoff time to start time AND endTime to latePickup time. Would storing this as DateTime (with a constant Date) make things easier in the future? CC: @annechung @AmandaMisjuwar

**Review by dhmi19 — CHANGES_REQUESTED** [review]
Looks good to me overall. Just left a few minor comments. I think there will be a few design changes needed as designers need to fix the Frame width and adjust the designs accordingly (everything looks really big right now). We also forgot to display the EDLP timings on the camp level info so we'll need to add that too (although, I think we can merge this PR if it gets blocked by those design changes)

File: frontend/src/constants/AuthConstants.ts
Comment (AmandaMisjuwar) [reply frontend/src/constants/AuthConstants.ts:5]: We need this line for other tickets as well so it'll be fine to leave

Comment (dhmi19) [reply frontend/src/constants/AuthConstants.ts:5]: sounds good!

File: frontend/src/components/pages/CampOverview/index.tsx
Comment (AmandaMisjuwar) [inline frontend/src/components/pages/CampOverview/index.tsx:29]: Can we call this `CampStatus` instead so not to confuse it with `UserStatus` in otherr components?

File: frontend/src/types/CampsTypes.ts
Comment (AmandaMisjuwar) [inline frontend/src/types/CampsTypes.ts:1]:
```suggestion
export type CampResponse = {
```

File: frontend/src/APIClients/CampsAPIClient.ts
Comment (AmandaMisjuwar) [inline frontend/src/APIClients/CampsAPIClient.ts:12]: Can we actually call this CampResponse instead so we can stay consistent with our types for all the backend responses/return values?

```suggestion
    return error as CampResponse;
```

File: frontend/src/components/pages/CampOverview/index.tsx
Comment (AmandaMisjuwar) [inline frontend/src/components/pages/CampOverview/index.tsx:39]: We should have some kind of error handling for if campResponse is `null`
```suggestion
      if (campResponse) setCamp(campResponse);
      else // display an error or just let it remain null
```

File: frontend/src/components/pages/CampOverview/index.tsx
Comment (AmandaMisjuwar) [inline frontend/src/components/pages/CampOverview/index.tsx:51]: related to the previous comment about handling null campResponse/errors, we won't need the ? potentially

File: frontend/src/types/CampsTypes.ts
Comment (AmandaMisjuwar) [reply frontend/src/types/CampsTypes.ts:10]: Yea if we need to do date/time calculations we should store them as Dates (however in Mongo they're currently also strings). But if they're always 30 min time slots we could also just convert from string -> Date -> string whenever we do the calculation? @annechung @dhmi19

**Review by AmandaMisjuwar — CHANGES_REQUESTED** [review]

File: frontend/src/App.tsx
Comment (annechung) [reply frontend/src/App.tsx:109]: yeah we can remove this

File: frontend/src/types/CampsTypes.ts
Comment (annechung) [reply frontend/src/types/CampsTypes.ts:10]: I feel like it makes more sense to store as strings since these fields are actually times that aren’t tied to a specific date?

Comment (AmandaMisjuwar) [reply frontend/src/types/CampsTypes.ts:10]: Yeah we'll just have to make sure theyre in a format that can be converted into Date if needed for calculation (will leave it as string for now)

**Review by annechung — APPROVED** [review]
cannot wait to overview all the camps

**Review by AmandaMisjuwar — APPROVED** [review]

**Pull request merged**
